transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DialectH2Impl.java

```diff
@@ -0,0 +1,7 @@
+package com.gruelbox.transactionoutbox;
+
+import lombok.EqualsAndHashCode;
+
+/** Dialect SQL implementation for H2. */
+@EqualsAndHashCode
+public class DialectH2Impl extends DialectMySQL5Impl {}
```

Comment:
The use of inheritance here fails the "is a" test.

Since `DialectMySQQL5Impl` is a public class, it forms part of the API of the library, so we have to be careful here.

I'd be much happier if all the `Dialect` implementations were `final`, and composition/forwarding were used for code sharing. Happier still if the classes were also package-private, so they had no API footprint. There's no need for them to be public, so let's not expose them.

Reply:
Ok, took another stab at this. Implementations are now package private and final. Added new focused base implementations thinking it was easier to understand what functionality was supported but let me know if you were thinking to just use one of the existing implementations.